lib/timers.js

```diff
@@ -453,11 +452,9 @@ function setTimeout(callback, after, arg1, arg2, arg3) {
 }
 
 setTimeout[internalUtil.promisify.custom] = function(after, value) {
-  const promise = createPromise();
-  const timeout = new Timeout(promise, after, [value], false);
-  active(timeout);
-
-  return promise;
+  return new Promise((resolve) => {
+    active(new Timeout(resolve, after, [value], false));
```

Comment:
I don't think `active` can throw but if it can this is also a behavior change. In any case maybe this can just be `return new Promise(resolve => setTimeout(resolve, after, value))`?